src/AutoRest.CSharp/Mgmt/Models/RequestPath.cs

```diff
@@ -70,50 +68,39 @@ internal struct RequestPath : IEquatable<RequestPath>, IReadOnlyList<Segment>
             new Segment(new Reference("managementGroupId", typeof(string)), true, false)
         });
 
-        private IReadOnlyList<Segment> _segments;
+        private readonly IReadOnlyList<Segment> _segments;
 
-        /// <summary>
-        /// Constructs the <see cref="RequestPath"/> instance using the information in a <see cref="RestClientMethod"/>
-        /// Only the information about request path is used, other information, like HttpMethod is not used.
-        /// This is based on the facts that the paths of different operations under the same path are the same - they could have different non-path parameters, but path parameters must be the same
-        /// </summary>
-        /// <param name="method"></param>
-        public RequestPath(RestClientMethod method)
-            : this(method.Request.PathSegments, method.Operation.GetHttpPath())
+        public static RequestPath FromOperation(Operation operation, OperationGroup operationGroup)
         {
-        }
+            foreach (var request in operation.Requests)
+            {
+                var httpRequest = request.Protocol.Http as HttpRequest;
+                if (httpRequest is null)
+                    continue;
 
-        public static RequestPath FromPathSegments(PathSegment[] pathSegments, string httpPath)
-        {
-            return new RequestPath(pathSegments, httpPath);
+                var references = new MgmtRestClientBuilder(operationGroup).GetReferencesToOperationParameters(operation, request.Parameters);
```

Comment:
`MgmtRestClientBuilder` can be shared between `operationGroup` instances, but `operationGroup.Key` is used in the exception below.